### PR TITLE
Fix #12700.

### LINKED ISSUE #12700: `"doc" :at-m` can't document imported macros
```julia
module M
  macro m() end
end
```
```julia
julia> import M.@m

julia> "macro m"
       :@m
ERROR: Main.@m not found.
 in error at error.jl:21
```

This worked fine a few weeks ago. Current workaround:
```julia
julia> "macro m"
       :(M.@m)

help?> M.@m
  macro m
```

**Edit**: version info:
```
Julia Version 0.4.0-dev+6827
Commit e5e8ed5 (2015-08-19 13:22 UTC)
```

### PR DESCRIPTION
**Pull request by MichaelHatherly**

### CODE REVIEW

**Comment by MichaelHatherly** [comment]
Test case added as well now.

File: base/docs/bindings.jl [hunk 1]
```diff
@@ -6,7 +6,7 @@ moduleusings(mod) = ccall(:jl_module_usings, Any, (Any,), mod)
 
 function findsource(mod::Module, var::Symbol, seen = Set{Module}())
     mod in seen && return
-    var in names(mod, true) && return mod
+    var in names(mod, true, true) && return mod
```
Comment (vtjnash) [inline base/docs/bindings.jl]: would it be equivalent and much more efficient to use `isdefined(mod, var)` here?

Comment (MichaelHatherly) [inline base/docs/bindings.jl]: Yes, I believe so. Definitely more efficient as well.

**Comment by MichaelHatherly** [comment]
@one-more-minute, there's some other places where ``findsource`` is still failing to locate the correct module, but it's kind of late now so I'll have a look at it again tomorrow morning.

**Comment by one-more-minute** [comment]
Yeah, on second thought this will always report the given module as the source of the definition.

There must be a lower-level way to say "where did this binding come from"; we just need to figure out how to hook into it.

**Comment by stevengj** [comment]
In general, I would prefer more descriptive commit messages, so that I can read the log (or the PR) and see what is for without having to chase a link to another issue.

**Comment by MichaelHatherly** [comment]
> I would prefer more descriptive commit messages

Sure, will do.

> where did this binding come from

At least for macros we can get at the ``.code.module`` field to give us that, but ``typealias`` and ``const`` are still problematic...

**Comment by one-more-minute** [comment]
As an aside, even though it'd be nice to flesh out all possible edge cases here, documenting a macro or const from outside of its own module seems like a strange thing to want to do in the first place. I think it can be safely marked as low-priority (for the sake of any releases that might happen to be coming up).

**Comment by MichaelHatherly** [comment]
> seems like a strange thing to want to do in the first place. I think it can be safely marked as low-priority

Totally agree with that. I'm going to close this PR for the moment, since it's not *really* fixing the problem, and we can just track it in #12700.

**Comment by fcard** [comment]
It's pretty strange, yeah, and there is a workaround for it, so it's really not a big deal, thanks for tackling this anyway. It was "necessary" for my own version of not having to write docs and docstrings twice ([as seen here](https://github.com/fcard/ExpressionPatterns.jl/blob/master/src/Docs/makedocs.jl); [example](https://github.com/fcard/ExpressionPatterns.jl/blob/master/src/Docs/Dispatch.jl)), but I can adapt it for now (or I could try fixing this myself, if there is no problem).

**Comment by MichaelHatherly** [comment]
> or I could try fixing this myself

As @one-more-minute mentioned, it's not terribly high priority at the moment, but having more people get to know the ``Docs`` module would be good I think.

(btw, that's a cool package)

**Comment by fcard** [comment]
Alrighty, I will try finding a solution, and once everybody is less busy I will submit a PR.

(thanks!)

**Comment by StefanKarpinski** [comment]
On Wednesday, August 19, 2015, Steven G. Johnson <notifications@github.com>
wrote:

> In general, I would prefer more descriptive commit messages, so that I can
> read the log (or the PR) and see what is for without having to chase a link
> to another issue.
>

THIS. I *hate* this. Every commit message should have a brief description
of what it does and the body should have brief justification (unless it is
completely obvious). This minimalist commit message business a terrible
habit.

>

**Pull request closed without merging**